cloudpickle/cloudpickle.py

```diff
@@ -95,6 +95,12 @@
     PY2 = False
     from importlib._bootstrap import _find_spec
 
+_extract_code_globals_cache = (
+    weakref.WeakKeyDictionary()
+    if not hasattr(sys, "pypy_version_info")
+    else {})
```

Comment:
Unrelated to this PR review, but out of curiosity, do you know why we don't use a `weakref.WeakKeyDictionary()` under PyPy?

Reply:
Because they cannot create references to builtin-code objects. Again, this should not happen in the first place because builtin funcs should not be pickled dynamically (and thus should not have their globals extracted)

Reply:
Ah ok. Could you please add a comment to explain this?

Maybe with a `XXX` to explain that this should be refactored at some point to make it explicit that the extract globals function should never be called on functions/methods with builtin-code objects.

Reply:
OK. I gave it a try in #278.

Reply:
Great comment, thank you very much.